nck/readers/mytarget_reader.py

```diff
@@ -0,0 +1,408 @@
+        self.set_agency_client_token(refreshed_token)
+
+    def __retrieve_all_data(
+        self) -> Tuple[
+            Dict[str, List[Dict[str, Any]]],
+            List[Dict[str, str]],
+            Dict[str, str],
+            Dict[str, str]]:
+        response_id = self.__get_all_results('get_campaign_ids')
+        response_name = self.__get_all_results('get_campaign_names')
+
+        names_dict = self.convert_list_dicts_to_dict(response_name, 'name')
+        ids_dict = self.convert_list_dicts_to_dict(response_id, 'campaign_id')
+
+        campaign_ids = [element['campaign_id'] for element in response_id]
+        rsp_banner = self.get_all_banners_all_camp(campaign_ids)
+
+        rsp_daily_stat = self.__get_response('get_banner_stats', banner_ids=rsp_banner.keys())
+        rsp_banner_names = self.get_banner_name_response('get_banner_names', list(rsp_banner))
+
+        return rsp_daily_stat, names_dict, ids_dict, rsp_banner_names
+
+    def __get_all_results(
+        self,
+        name_content: str,
+        offset=0,
+        campaign_id=0,
+        banner_ids=[]
+    ) -> List[Dict[str, str]]:
+        """Based on the __get_response function this function is incrementing through offsets according to the
+        number of elements given by the first response.
+
+        Args:
+            name_content (str): string representing key of parameters config dict
+            offset (int, optional): potential offset of the request. Defaults to 0.
+            campaign_id (int, optional): potential campaign id of the request. Defaults to 0.
+            banner_ids (list, optional): potential banner ids list of the request. Defaults to [].
+
+        Returns:
+            List[Dict[str, Any]]: list of dicts resulting from the requests we made to the api
+        """
+        first_elements = self.__get_response(name_content, campaign_id=campaign_id, banner_ids=banner_ids)
+        count = first_elements['count']
+        elements = [first_elements['items']]
+        if count > 20:
+            elements += [
+                self.__get_response(name_content, offset=offset, campaign_id=campaign_id, banner_ids=banner_ids)['items']
+                for offset in range(20, self.round_up_to_base(count, 20), 20)
+            ]
+        return list(itertools.chain.from_iterable(elements))
+
+    def get_all_banners_all_camp(self, campaign_ids, banner_ids=[]):
+        all_banner = []
+        for campaign_id in campaign_ids:
+            all_banner.extend(self.get_all_banners_one_camp(campaign_id, banner_ids=banner_ids))
+        return dict(all_banner)
+
+    def get_all_banners_one_camp(self, campaign_id, banner_ids=[]):
+        all_banners = self.__get_all_results('get_banner_ids', campaign_id=campaign_id, banner_ids=banner_ids)
+        filtered_banners = [(ban['id'], ban['campaign_id']) for ban in all_banners if ban['moderation_status'] == 'allowed']
+        return filtered_banners
```

Comment:
(optional) `get_all_banners_one_camp` could return a generator. Thus, we would have been able to do a simple list comprehension in `get_all_banners_all_camp`. It would have been more efficient and easily readable.